packages/lexical-list/src/formatList.ts

```diff
@@ -400,6 +400,8 @@ export function $handleIndent(listItemNode: ListItemNode): void {
       const newListItem = $createListItemNode()
         .setTextFormat(parent.getTextFormat())
         .setTextStyle(parent.getTextStyle());
+      // Copy the text style from the list item being indented to preserve marker styles
+      newListItem.setTextStyle(listItemNode.getTextStyle());
```

Comment:
Looking more closely I think the fix can be even simpler, the text style is just coming from the wrong node. No need to set it twice. It does make me think that both format and style should probably come from the same node though

```suggestion
        .setTextFormat(listItemNode.getTextFormat())
        .setTextStyle(listItemNode.getTextStyle());
```

Reply:
Thanks for catching that @etrepum! I've [simplified the code](https://github.com/facebook/lexical/pull/7504/commits/d3f20dc10d70fa186a1afafeafb8bb96a561fe86) to:
1. Get both format and style from the listItemNode being indented
2. Remove the redundant style setting
3. Make the format and style handling consistent

Thanks for the thorough review. Please let me know if this looks better!